src/contentmanager.cpp

```diff
@@ -108,12 +112,29 @@ void ContentManager::onCustomContextMenu(const QPoint &point)
         }
         contextMenu.addAction(&menuCancelBook);
     } else {
-        if (m_local) {
+        try {
+            const auto book = KiwixApp::instance()->getLibrary()->getBookById(id);
+            auto bookPath = QString::fromStdString(book.getPath());
             contextMenu.addAction(&menuOpenBook);
             contextMenu.addAction(&menuDeleteBook);
-        }
-        else
+            contextMenu.addAction(&menuOpenFolder);
+            connect(&menuOpenFolder, &QAction::triggered, [=]() {
+                QFileInfo fileInfo(bookPath);
+                QDir bookDir = fileInfo.absoluteDir();
+                bool dirOpen = bookDir.exists() && bookDir.isReadable() && QDesktopServices::openUrl(bookDir.absolutePath());
```

Comment:
@juuz0 Here is quite comment code pattern and let me take a few minutes to make an in-depth analysis.

You have two different things you might be interested to know:

1. * Does the directory can be "open" (boolean answer)
2. * Maybe, if it fails: Why it fails to open

My request was only about (1) and to do that you should only rely on `QDesktopServices::openUrl` IMO.

Then, you might want to do (2) to help the user with a meaningful message (why the folder can't actually be open), but here you don't as your message is always the same.

Would you be able to implement it that way, or do the multiple checks on line 124 are here to compensate some kind of weakness of the `QDesktopServices::openUrl` (for example it open even if the directory is not readable?)

Reply:
> do the multiple checks on line 124 are here to compensate..........

@kelson42 
Yes, this is an issue with [`QDesktopServices::openUrl`](https://doc.qt.io/qt-6/qdesktopservices.html#openUrl) 
From the article:
> Warning: A return value of true indicates that the application has successfully requested the operating system to open the URL in an external application. The external application may still fail to launch or fail to open the requested URL. This result will not be reported back to the application.

Should put a better message yeah, should these be enough?

1. Kiwix is not able to open the location because the folder doesn't exist
2. Kiwix is not able to open the location because the folder is not readable
3. Kiwix is not able to open the location [Generic]

Reply:
> Could we let the file explorer display a message to the user (it is better placed to know why it has fail to open the directory and so gives a useful message).

This works for cases where directory is not readable.
But atleast on Ubuntu, if the directory doesn't exist - file explorer wont launch but show `gio: No such file or directory.` in the terminal. 
Most people won't have kiwix-desktop running through a terminal so atleast for the case where `directory doesn't exist`, its better to show the dialog in kiwix-desktop?

Reply:
Is it possible that we have a zim file displayed in the library view when the containing directory doesn't exist ?

Reply:
I doubt it.

ContentManager use a filter with local and valid (https://github.com/kiwix/kiwix-desktop/blob/main/src/contentmanager.cpp#L692-L694).
And valid means that we can access the file : (I spare you all the code, but at this end it is : https://github.com/kiwix/libkiwix/blob/main/src/tools/pathTools.cpp#L314-L321)

This could be possible if the user remove the directory after loading the library. We should be prepare for this (not crashing) but I'm not sure we have to do more.
And if we have to do more, we should probably doing it by detecting this sooner and "greying" the book in the view and not display a "open in directory".

Reply:
> @kelson42 
> Yes, this is an issue with [`QDesktopServices::openUrl`](https://doc.qt.io/qt-6/qdesktopservices.html#openUrl) 

Then fine to me. Current approach is probably the best compromise. Most important is not to fail silently.